src/health.py

```diff
@@ -0,0 +1,170 @@
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
+        )
+
+    def _get_partitions_size(self) -> Tuple[int, int]:
+        """Gets the number of partitions and their average size from the log dirs."""
+        log_dirs_command = [
+            "--describe",
+            f"--bootstrap-server {','.join(self.charm.kafka_config.bootstrap_server)}",
+            f"--command-config {self.charm.kafka_config.client_properties_filepath}",
+        ]
+        log_dirs = self.charm.snap.run_bin_command(
+            bin_keyword="log-dirs", bin_args=log_dirs_command
+        )
+
+        dirs = {}
+        for line in log_dirs.splitlines():
+            try:
+                # filters stdout to only relevant lines
+                dirs = json.loads(line)
+                break
+            except json.decoder.JSONDecodeError:
+                continue
+
+        if not dirs:
+            return (0, 0)
+
+        partitions = []
+        sizes = []
+        for broker in dirs["brokers"]:
+            for log_dir in broker["logDirs"]:
+                for partition in log_dir["partitions"]:
+                    partitions.append(partition["partition"])
+                    sizes.append(int(partition["size"]))
+
+        average_partition_size = mean(sizes)
+        total_partitions = len(partitions)
+
+        return (total_partitions, average_partition_size)
+
+    def _check_memory_maps(self) -> bool:
+        """Checks that the number of used memory maps is not approaching threshold."""
+        max_maps = self._get_max_memory_maps()
+        current_maps = self._get_current_memory_maps()
+
+        # eyeballing warning if 80% used, can be changed
+        if max_maps * 0.8 <= current_maps:
+            self.charm._set_status(Status.NEAR_MMAP_LIMIT)
+            return False
+
+        return True
+
+    def _check_file_descriptors(self) -> bool:
+        """Checks that the number of used file descriptors is not approaching threshold."""
+        if not self.charm.kafka_config.client_listeners:
+            return True
+
+        total_partitions, average_partition_size = self._get_partitions_size()
+        segment_size = int(self.charm.config["log_segment_bytes"])
+
+        minimum_fd_limit = total_partitions * (average_partition_size / segment_size)
+        current_max_files = self._get_current_max_files()
+
+        # eyeballing warning if 80% used, can be changed
+        if current_max_files * 0.8 <= minimum_fd_limit:
+            self.charm._set_status(Status.NEAR_FD_LIMIT)
+            return False
+
+        return True
+
+    def _check_vm_swappiness(self) -> bool:
+        """Checks that vm.swappiness is configured correctly on the machine."""
+        vm_swappiness = self._get_vm_swappiness()
+
+        if vm_swappiness > 1:
+            self.charm._set_status(Status.TOO_SWAPPY)
+            return False
+
+        return True
+
+    def machine_configured(self) -> bool:
```

Comment:
**niptick** `is_machine_configured()`

I know it is a small thing but the "is_something" actually makes this very readable that is going to be a boolean (as the signature shows)

Reply:
I agree sortof, but I think it can go both ways depending on the circumstance. If we assume we're referencing with the following:

```python
if Object.bool_func():
# or
if not Object.bool_func():
```

If you have an object, e.g `Animal` with a method that checks if that person is a cat, it would be appropriate to have an `is_cat`, as the `is_` prefix is coupled to the `Animal` reference. So the 'sentence' becomes 'if the animal is a cat':
```python
# sounds good
if Animal.is_cat():
# sounds bad
if Animal.cat():
```

Here however, the 'sentence' is 'if the charm health configured correctly', which I don't think sounds bad. I think the sentence 'if the charm health is configured correctly' sounds wrong. I think the possible difference is because 'configured' is a past tense verb being used for the noun 'machine' rather than the noun 'charm' or 'health'.
'is_machine_configured' sounds like a question, 'is_configured' sounds like an adjective.

I would propose instead to rename `health.KafkaHealth` to `machine.KafkaMachine`, and THEN rename this method to `is_configured`. That way the sentence reads 'if machine is configured', which sounds nice.